docling_core/types/experimental/document.py

```diff
@@ -0,0 +1,1106 @@
+                            page_w=page_w,
+                            page_h=page_h,
+                            xsize=xsize,
+                            ysize=ysize,
+                            page_i=self.prov[0].page_no,
+                        )
+                    elif (
+                        col.bbox is not None
+                        and add_cell_location
+                        and not add_page_index
+                        and len(self.prov) > 0
+                    ):
+                        page_w, page_h = doc.pages[self.prov[0].page_no].size.as_tuple()
+
+                        cell_loc = DocumentToken.get_location(
+                            bbox=col.bbox.to_bottom_left_origin(page_h).as_tuple(),
+                            page_w=page_w,
+                            page_h=page_h,
+                            xsize=xsize,
+                            ysize=ysize,
+                            page_i=-1,
+                        )
+
+                    cell_label = ""
+                    if add_cell_label:
+                        if col.column_header:
+                            cell_label = "<col_header>"
+                        elif col.row_header:
+                            cell_label = "<row_header>"
+                        elif col.row_section:
+                            cell_label = "<row_section>"
+                        else:
+                            cell_label = "<body>"
+
+                    body += f"<col_{j}>{cell_loc}{cell_label}{text}</col_{j}>"
+
+                body += f"</row_{i}>{new_line}"
+
+        body += f"{DocumentToken.END_TABLE.value}{new_line}"
+
+        return body
+
+
+class KeyValueItem(DocItem):
+    """KeyValueItem."""
+
+
+ContentItem = Union[TextItem, FigureItem, TableItem, KeyValueItem]
+
+
+class DocumentTrees(BaseModel):
+    """DocumentTrees."""
+
+    furniture: GroupItem = GroupItem(
+        name="_root_", dloc="#/furniture"
+    )  # List[RefItem] = []
+    body: GroupItem = GroupItem(name="_root_", dloc="#/body")  # List[RefItem] = []
+
+
+class PageItem(BaseModel):
+    """PageItem."""
+
+    # A page carries separate root items for furniture and body,
+    # only referencing items on the page
+    hash: str  # page hash
+    size: Size
+    image: Optional[ImageRef] = None
+    page_no: int
+
+
+class DoclingDocument(DocumentTrees):
+    """DoclingDocument."""
+
+    version: str = "0.0.1"  # = SemanticVersion(version="0.0.1")
+    description: Any
+    file_info: FileInfo
+
+    groups: List[GroupItem] = []
+    texts: List[TextItem] = []
+    figures: List[FigureItem] = []
+    tables: List[TableItem] = []
+    key_value_items: List[KeyValueItem] = []
+
+    pages: Dict[int, PageItem] = {}  # empty as default
+
+    def add_group(
+        self,
+        label: Optional[GroupLabel] = None,
+        name: Optional[str] = None,
+        parent: Optional[GroupItem] = None,
```

Comment:
Do we really want to allow the case of label, name, and parent all being None?

Reply:
Probably no. There must be a parent always. `label` has a default, so users don't need to provide it. `name` is optional.

Reply:
`parent` is using `self.body` as default if unset.